ledger-api/grpc-definitions/com/digitalasset/ledger/api/v1/event.proto

```diff
@@ -111,6 +113,87 @@ message ArchivedEvent {
   repeated string witness_parties = 4;
 }
 
+// Records that a contract has entered the view of the participant.
+// The contract has been created on a domain on which the participant does not represent
+// all stakeholders of the contract or has previously left the view.
+// Used only for partitioned ledger topologies with multiple domains.
+message EnteredViewEvent {
+  // The ID of this particular event.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string event_id = 1;
+
+  // The ID of the contract.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Required
+  string contract_id = 2;
+
+  // The template of the contract.
+  // Required
+  Identifier template_id = 3;
+
+  // The key of the contract, if defined.
+  // Optional
+  Value contract_key = 7;
+
+  // The arguments that have been used to create the contract.
+  // Required
+  Record create_arguments = 4;
```

Comment:
It might be premature to talk about this already now, but I would assume that at some point, we might want to avoid sending the payload several times and therefore split the payload from status updates and only send the payload once.